File: src/engine/commands/admin/admincommand.cpp
#include "admincommand.h"

#include "realm.h"


#define super Command

AdminCommand::AdminCommand(QObject *parent) :
    super(parent) {
}

AdminCommand::~AdminCommand() {
}

GameObjectPtrList AdminCommand::objectsByDescription(const ObjectDescription &description,
                                                     const GameObjectPtrList &pool) {

    if (description.name == "room") {
        GameObjectPtrList objects;
        objects << currentRoom();
        return objects;
    }

    if (description.name.startsWith("#")) {
        uint objectId = description.name.mid(1).toInt();
        if (objectId > 0 || description.name == "#0") {
            GameObjectPtrList objects;
            if (description.position < 2) {
                objects << realm()->getObject(GameObjectType::Unknown, objectId);
            }
            return objects;
        }
    }

    return super::objectsByDescription(description, pool);
}

File: src/engine/commands/admin/admincommand.js

function AdminCommand() {
    Command.call(this);
}

AdminCommand.prototype = new Command();
AdminCommand.prototype.constructor = AdminCommand;

AdminCommand.prototype.objectsByDescription = function(description, pool) {

    if (description.name === "room") {
        return [ this._currentRoom ];
    }

    if (description.name.startsWith("#")) {
        var objectId = description.name.mid(1).toInt();
        if (!isNaN(objectId)) {
            var objects = [];
            if (description.position < 2) {
                objects.append(Realm.getObject("Unknown", objectId));
            }
            return objects;
        }
    }

    return Command.prototype.objectsByDescription.call(this, description, pool);
};
